projects/can_transceiver/src/can_transceiver_ros_intf.cpp

```diff
@@ -44,12 +60,117 @@ class CanTransceiverIntf : public rclcpp::Node
+        pub_filtered_wind_sensor_ =
+          this->create_publisher<custom_interfaces::msg::WindSensor>(FILTERED_WIND_SENSOR_TOPIC, QUEUE_SIZE);
+        pub_wind_sensors_ = this->create_publisher<custom_interfaces::msg::WindSensors>(WIND_SENSORS_TOPIC, QUEUE_SIZE);
+        pub_batteries_    = this->create_publisher<custom_interfaces::msg::Batteries>(BATTERIES_TOPIC, QUEUE_SIZE);
+        pub_data_sensors_ =
+          this->create_publisher<custom_interfaces::msg::GenericSensors>(DATA_SENSORS_TOPIC, QUEUE_SIZE);
+        // Timer with 500ms delay
+        timer_ = this->create_wall_timer(
+          std::chrono::milliseconds(PLACEHOLDER_MS), std::bind(&CanSimIntf::timer_callback, this));
+    }
+
+private:
+    // Subscriber
+    void gps_callback(const custom_interfaces::msg::GPS::SharedPtr msg) const {}
+    void wind_callback(const custom_interfaces::msg::WindSensors::SharedPtr msg) const {}
+
+    // Publisher
+    void timer_callback()
+    {
+        // ** GPS **
+        custom_interfaces::msg::GPS gps_data;
+        gps_data.heading.heading   = PLACEHOLDER_VALUE;
+        gps_data.speed.speed       = PLACEHOLDER_VALUE;
+        gps_data.lat_lon.latitude  = PLACEHOLDER_VALUE;
+        gps_data.lat_lon.longitude = PLACEHOLDER_VALUE;
+        // Publishes msg
+        pub_gps_->publish(gps_data);
+
+        // ** Filtered Wind Sensor **
+        custom_interfaces::msg::WindSensor filtered_wind_sensor_data;
+        filtered_wind_sensor_data.speed.speed = PLACEHOLDER_VALUE;
+        filtered_wind_sensor_data.direction   = PLACEHOLDER_VALUE;
+        // Publishes msg
+        pub_filtered_wind_sensor_->publish(filtered_wind_sensor_data);
+
+        // ** Wind Sensors **
+        custom_interfaces::msg::WindSensors wind_sensors_data;
+        for (auto & sensor : wind_sensors_data.wind_sensors) {
+            sensor.direction   = PLACEHOLDER_VALUE;
+            sensor.speed.speed = PLACEHOLDER_VALUE;
+        }
+        // Publishes msg
+        pub_wind_sensors_->publish(wind_sensors_data);
+
+        // ** Batteries **
+        custom_interfaces::msg::Batteries batteries_data;
+        for (auto & sensor : batteries_data.batteries) {
+            sensor.voltage = PLACEHOLDER_VALUE;
+            sensor.current = PLACEHOLDER_VALUE;
+        }
+        // Publishes msg
+        pub_batteries_->publish(batteries_data);
+
+        // ** Data Sensors **
+        custom_interfaces::msg::HelperGenericSensor helper_gen_sensor_data;
+        custom_interfaces::msg::GenericSensors      gen_sensor_data;
+        helper_gen_sensor_data.id   = 0x0;  //uint8
+        helper_gen_sensor_data.data = 0x0;  //uint64
+        // Publishes msg
+        gen_sensor_data.generic_sensors.push_back(helper_gen_sensor_data);
+        pub_data_sensors_->publish(gen_sensor_data);
+    }
+
+    // Subscriber Field Declarations
+    rclcpp::Subscription<custom_interfaces::msg::GPS>::SharedPtr         sub_gps_;
```

Comment:
Small nitpick. Change sub_gps_ to sub_mock_gps_

Reply:
gps -> mock gps when subscribing